sklearn/neighbors/tests/test_nearest_centroid.py

```diff
@@ -142,6 +146,20 @@ def test_manhattan_metric():
     assert_array_equal(dense_centroid, [[-1, -1], [1, 1]])
 
 
+# TODO(1.5): remove this test
+@pytest.mark.parametrize(
+    "metric", list(NearestCentroid._valid_metrics - {"manhattan", "euclidean"})
+)
+def test_deprecated_metrics(metric):
+    # Check that a warning is raised for all deprecated metrics
```

Comment:
This should fix the failure on the CI (the iterable for `metric` must be the same across pytest workers). Also some nitpicks.

```suggestion
    "metric", sorted(list(NearestCentroid._valid_metrics - {"manhattan", "euclidean"}))
)
def test_deprecated_distance_metric_supports(metric):
    # Check that a warning is raised for all deprecated distance metric supports
```